From d9a3e3ef8aa2b4fb42a448e5a96ad0f163b6fa65 Mon Sep 17 00:00:00 2001
From: Ethan Jackson <ethan@nicira.com>
Date: Tue, 21 Sep 2010 18:03:07 -0700
Subject: [PATCH 13/22] xenserver: Rename monitor-external-ids -> ovs-external-ids

Renamed the monitor-external-ids script ovs-external-ids.
Hopefully this will make it clearer who owns it when someone does
ps xa.

Also removed trailing whitespace from ovs-external-ids.

Signed-off-by: Ethan Jackson <ethan@nicira.com>
---
 xenserver/README                                   |    2 +-
 xenserver/automake.mk                              |    2 +-
 xenserver/etc_init.d_openvswitch                   |    6 ++--
 xenserver/openvswitch-xen.spec                     |    6 ++--
 ...usr_share_openvswitch_scripts_ovs-external-ids} |   34 +++++++++----------
 5 files changed, 24 insertions(+), 26 deletions(-)
 rename xenserver/{usr_share_openvswitch_scripts_monitor-external-ids => usr_share_openvswitch_scripts_ovs-external-ids} (96%)

diff --git a/xenserver/README b/xenserver/README
index 9fcdb1a..8b2b926 100644
--- a/xenserver/README
+++ b/xenserver/README
@@ -49,7 +49,7 @@ files are:
 
         Open vSwitch-aware replacement for Citrix script of the same name.
 
-    usr_share_openvswitch_scripts_monitor-external-ids
+    usr_share_openvswitch_scripts_ovs-external-ids
 
         Daemon to monitor the external_ids columns of the Bridge and
         Interface OVSDB tables.
diff --git a/xenserver/automake.mk b/xenserver/automake.mk
index ffd5996..21cba5d 100644
--- a/xenserver/automake.mk
+++ b/xenserver/automake.mk
@@ -23,7 +23,7 @@ EXTRA_DIST += \
 	xenserver/usr_lib_xsconsole_plugins-base_XSFeatureVSwitch.py \
 	xenserver/usr_sbin_brctl \
 	xenserver/usr_sbin_xen-bugtool \
-	xenserver/usr_share_openvswitch_scripts_monitor-external-ids \
+	xenserver/usr_share_openvswitch_scripts_ovs-external-ids \
 	xenserver/usr_share_openvswitch_scripts_refresh-xs-network-uuids \
 	xenserver/usr_share_openvswitch_scripts_sysconfig.template \
 	xenserver/uuid.py
diff --git a/xenserver/etc_init.d_openvswitch b/xenserver/etc_init.d_openvswitch
index 68079fc..d54a18f 100755
--- a/xenserver/etc_init.d_openvswitch
+++ b/xenserver/etc_init.d_openvswitch
@@ -346,7 +346,7 @@ function start {
 
     # Start daemon to monitor external ids
     PYTHONPATH=/usr/share/openvswitch/python \
-               /usr/share/openvswitch/scripts/monitor-external-ids \
+               /usr/share/openvswitch/scripts/ovs-external-ids \
                --pidfile --detach "$VSWITCHD_OVSDB_SERVER"
 
     touch /var/lock/subsys/openvswitch
@@ -356,8 +356,8 @@ function stop {
     stop_daemon BRCOMPATD "$brcompatd"
     stop_daemon VSWITCHD "$vswitchd"
     stop_daemon OVSDB_SERVER "$ovsdb_server"
-    if [ -e /var/run/openvswitch/monitor-external-ids.pid ]; then
-        kill `cat /var/run/openvswitch/monitor-external-ids.pid`
+    if [ -e /var/run/openvswitch/ovs-external-ids.pid ]; then
+        kill `cat /var/run/openvswitch/ovs-external-ids.pid`
     fi
     rm -f /var/lock/subsys/openvswitch
 }
diff --git a/xenserver/openvswitch-xen.spec b/xenserver/openvswitch-xen.spec
index e420ec6..90fee2a 100644
--- a/xenserver/openvswitch-xen.spec
+++ b/xenserver/openvswitch-xen.spec
@@ -73,8 +73,8 @@ install -m 644 xenserver/opt_xensource_libexec_InterfaceReconfigureVswitch.py \
              $RPM_BUILD_ROOT/usr/share/openvswitch/scripts/InterfaceReconfigureVswitch.py
 install -m 755 xenserver/etc_xensource_scripts_vif \
              $RPM_BUILD_ROOT/usr/share/openvswitch/scripts/vif
-install -m 755 xenserver/usr_share_openvswitch_scripts_monitor-external-ids \
-               $RPM_BUILD_ROOT/usr/share/openvswitch/scripts/monitor-external-ids
+install -m 755 xenserver/usr_share_openvswitch_scripts_ovs-external-ids \
+               $RPM_BUILD_ROOT/usr/share/openvswitch/scripts/ovs-external-ids
 install -m 755 xenserver/usr_share_openvswitch_scripts_refresh-xs-network-uuids \
                $RPM_BUILD_ROOT/usr/share/openvswitch/scripts/refresh-xs-network-uuids
 install -m 755 xenserver/usr_sbin_xen-bugtool \
@@ -392,7 +392,7 @@ fi
 /usr/share/openvswitch/python/ovs/timeval.py
 /usr/share/openvswitch/python/ovs/util.py
 /usr/share/openvswitch/python/uuid.py
-/usr/share/openvswitch/scripts/monitor-external-ids
+/usr/share/openvswitch/scripts/ovs-external-ids
 /usr/share/openvswitch/scripts/refresh-xs-network-uuids
 /usr/share/openvswitch/scripts/interface-reconfigure
 /usr/share/openvswitch/scripts/InterfaceReconfigure.py
diff --git a/xenserver/usr_share_openvswitch_scripts_monitor-external-ids b/xenserver/usr_share_openvswitch_scripts_ovs-external-ids
similarity index 96%
rename from xenserver/usr_share_openvswitch_scripts_monitor-external-ids
rename to xenserver/usr_share_openvswitch_scripts_ovs-external-ids
index a0aad7a..40b7d6d 100755
--- a/xenserver/usr_share_openvswitch_scripts_monitor-external-ids
+++ b/xenserver/usr_share_openvswitch_scripts_ovs-external-ids
@@ -53,13 +53,13 @@ def init_session():
         session.xenapi.login_with_password("", "")
     except:
         session = None
-        syslog.syslog(syslog.LOG_WARNING, 
-                "monitor-external-ids: Couldn't login to XAPI")
+        syslog.syslog(syslog.LOG_WARNING,
+                "ovs-external-ids: Couldn't login to XAPI")
         return False
 
     return True
 
-# By default, the "bridge-id" external id in the Bridge table is the 
+# By default, the "bridge-id" external id in the Bridge table is the
 # same as "xs-network-uuids".  This may be overridden by defining a
 # "nicira-bridge-id" key in the "other_config" field of the network
 # record of XAPI.
@@ -73,7 +73,7 @@ def get_bridge_id(br_name, default=None):
             continue
         return rec['other_config'].get('nicira-bridge-id', default)
 
-# By default, the "iface-id" external id in the Interface table is the 
+# By default, the "iface-id" external id in the Interface table is the
 # same as "xs-vif-uuid".  This may be overridden by defining a
 # "nicira-iface-id" key in the "other_config" field of the VIF
 # record of XAPI.
@@ -101,15 +101,15 @@ def set_external_id(table, record, key, value):
     cmd = [vsctl, "--timeout=30", "-vANY:console:emer", "set", table, record, col]
     exitcode = subprocess.call(cmd)
     if exitcode != 0:
-        syslog.syslog(syslog.LOG_WARNING, 
-                "monitor-external-ids: Couldn't call ovs-vsctl")
+        syslog.syslog(syslog.LOG_WARNING,
+                "ovs-external-ids: Couldn't call ovs-vsctl")
 
 # XAPI on XenServer 5.6 uses the external-id "network-uuids" for internal
-# networks, but we now prefer "xs-network-uuids".  Look for its use and 
+# networks, but we now prefer "xs-network-uuids".  Look for its use and
 # write our preferred external-id.
 def update_network_uuids(name, ids):
     if ids["network-uuids"] and not ids["xs-network-uuids"]:
-        set_external_id("Bridge", name, "xs-network-uuids", 
+        set_external_id("Bridge", name, "xs-network-uuids",
                 ids["network-uuids"])
 
 def update_bridge_id(name, ids):
@@ -142,13 +142,13 @@ def keep_table_columns(schema, table_name, column_types):
         new_columns[column_name] = column
     table.columns = new_columns
     return table
- 
+
 def monitor_uuid_schema_cb(schema):
     string_type = types.Type(types.BaseType(types.StringType))
     string_map_type = types.Type(types.BaseType(types.StringType),
                                  types.BaseType(types.StringType),
                                  0, sys.maxint)
- 
+
     new_tables = {}
     for table_name in ("Bridge", "Interface"):
         new_tables[table_name] = keep_table_columns(
@@ -171,7 +171,7 @@ def main(argv):
     except getopt.GetoptError, geo:
         sys.stderr.write("%s: %s\n" % (ovs.util.PROGRAM_NAME, geo.msg))
         sys.exit(1)
- 
+
     for key, value in options:
         if key in ['-h', '--help']:
             usage()
@@ -179,14 +179,14 @@ def main(argv):
             sys.stderr.write("%s: unhandled option %s\n"
                              % (ovs.util.PROGRAM_NAME, key))
             sys.exit(1)
- 
+
     if len(args) != 1:
         sys.stderr.write("%s: exactly one nonoption argument is required "
                          "(use --help for help)\n" % ovs.util.PROGRAM_NAME)
         sys.exit(1)
 
     ovs.daemon.die_if_already_running()
- 
+
     remote = args[0]
     idl = ovs.db.idl.Idl(remote, "Open_vSwitch", monitor_uuid_schema_cb)
 
@@ -196,7 +196,6 @@ def main(argv):
     # tasks, we need it.  Wait here until it's up.
     while not os.path.exists("/var/run/xapi_init_complete.cookie"):
         time.sleep(1)
- 
     bridges = {}
     interfaces = {}
     while True:
@@ -205,7 +204,6 @@ def main(argv):
             idl.wait(poller)
             poller.block()
             continue
- 
         new_bridges = {}
         for rec in idl.data["Bridge"].itervalues():
             name = rec.name.as_scalar()
@@ -213,13 +211,13 @@ def main(argv):
             network_uuids = rec.external_ids.get("network-uuids")
             new_bridges[name] = {"xs-network-uuids": xs_network_uuids,
                                  "network-uuids": network_uuids}
- 
+
         new_interfaces = {}
         for rec in idl.data["Interface"].itervalues():
             name = rec.name.as_scalar()
             xs_vif_uuid = rec.external_ids.get("xs-vif-uuid")
             new_interfaces[name] = {"xs-vif-uuid": xs_vif_uuid}
- 
+
         if bridges != new_bridges:
             for name,ids in new_bridges.items():
                 # Network uuids shouldn't change in the life of a bridge,
@@ -237,7 +235,7 @@ def main(argv):
                 if (name not in interfaces) or (interfaces[name] != ids):
                     update_iface_id(name, ids)
             interfaces = new_interfaces
- 
+
 if __name__ == '__main__':
     try:
         main(sys.argv)
-- 
1.6.3.3

